Controller/RoleController.php

```diff
@@ -116,8 +117,15 @@ public function createRoleAction(Request $request)
      */
     public function updateRoleAction(Request $request, $roleId)
     {
-        $role = $this->roleService->loadRole($roleId);
-        $roleData = (new RoleMapper())->mapToFormData($role);
+        try {
+            $roleDraft = $this->roleService->loadRoleDraft($roleId);
+        } catch (NotFoundException $e) {
+            // The draft doesn't exist, let's create one
+            $role = $this->roleService->loadRole($roleId);
+            $roleDraft = $this->roleService->createRoleDraft($role);
+        }
+
+        $roleData = (new RoleMapper())->mapToFormData($roleDraft);
```

Comment:
This should rather be injected or otherwise registered to the form, but AFAICS this is not the case elsewhere as well, so there's probably a plan behind it :)